neural_speed/application/main_run.cpp

```diff
@@ -405,6 +405,43 @@ int main(int argc, char** argv) {  // NOLINT
   const bool penalize_nl = params.penalize_nl;
   model_token id = 0;
 
+  if (params.warmup) {
+    {
+      const std::vector<model_token> tmp(32, ctx->vocab.bos_token_id);
+      std::vector<model_input> inputs = {model_input{
+          /*.tokens              =*/tmp.data(),
+          /*.n_tokens           =*/(uint32_t)tmp.size(),
+          /*.n_prompt_tokens    =*/0,
+          /*.n_past             =*/0,
+          /*.n_total            =*/0,
+          /*.request_idx        =*/0,
+          /*.beam_idx           =*/0,
+          /*.padding_side       =*/0,
+          /*n_padding           =*/0,
+      }};
+      model_eval(ctx, inputs.data(), inputs.size(), params.n_threads);
+    }
+
+    {
+      const std::vector<model_token> tmp = {
+          0,
+      };
+      std::vector<model_input> inputs = {model_input{
+          /*.tokens              =*/tmp.data(),
+          /*.n_tokens           =*/(uint32_t)tmp.size(),
+          /*.n_prompt_tokens    =*/0,
+          /*.n_past             =*/(uint32_t)(params.n_predict - 1),
```

Comment:
Will it affect overall performance as warmup are added by default?

In addition, it causes invalid `n_past` if `-n` not set as command line arguments.

cc @zhenwei-intel

Reply:
The warmup time will not be counted. The timer will be cleared before real inference.

Reply:
`(uint32_t)(params.n_predict - 1)` can be replaced by 32.